server/storage/azure.go

```diff
@@ -0,0 +1,132 @@
+package storage
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/container"
+)
+
+type AzureStorage struct {
+	Storage
+	client          *azblob.Client
+	containerClient *container.Client
+	containerName   string
+	logger          *log.Logger
+}
+
+func getCredentials() (*azidentity.DefaultAzureCredential, error) {
+	return azidentity.NewDefaultAzureCredential(nil)
+}
+
+func NewAzureBlobStorage(ctx context.Context, storageAccountName string, containerName string, logger *log.Logger) (Storage, error) {
+	logger.Println("Creating Azure Blob Storage")
+
+	if storageAccountName == "" {
```

Comment:
you check for this on https://github.com/dutchcoders/transfer.sh/pull/638/files#diff-8e494a434a8037b6c0b888e25b2baae7618fe65e792d4a155dadd096e9350667R511. please do not repeat the check and leave it only on `cmd.go`